judge/models/problem.py

```diff
@@ -247,9 +247,9 @@ def clarifications(self):
         return ProblemClarification.objects.filter(problem=self)
 
     def update_stats(self):
-        self.user_count = self.submission_set.filter(points__gt=0).values('user').distinct().count()
+        self.user_count = self.submission_set.filter(points__gte=F('problem__points')).values('user').distinct().count()
```

Comment:
This is all the same problem, so you might as well just use `self.points` instead of that `F` non-sense?

Reply:
Also I think you should check `result='AC'` while you are at it, since 5.0/5 WA has happened. That is in fact how users who can see submissions are calculated.